lib/jellyfish/component/hls/request_handler.ex

```diff
@@ -0,0 +1,208 @@
+  Handles ll-hls partial requests
+  """
+  @spec handle_partial_request(Room.id(), String.t(), non_neg_integer()) ::
+          {:ok, binary()} | {:error | atom()}
+  def handle_partial_request(room_id, filename, offset) do
+    EtsHelper.get_partial(room_id, filename, offset)
+  end
+
+  @doc """
+  Handles manifest requests with specific partial requested (ll-hls)
+  """
+  @spec handle_manifest_request(Room.id(), partial()) :: {:ok, binary()} | {:error, any()}
+  def handle_manifest_request(room_id, partial) do
+    {:ok, last_partial} = EtsHelper.get_recent_partial(room_id)
+
+    unless is_partial_ready(partial, last_partial) do
+      wait_for_manifest_ready(room_id, partial, :manifest)
+    end
+
+    EtsHelper.get_manifest(room_id)
+  end
+
+  @doc """
+  Handles manifest requests with specific partial requested (ll-hls)
+  """
+  @spec handle_delta_manifest_request(Room.id(), partial()) :: {:ok, String.t()} | {:error, any()}
+  def handle_delta_manifest_request(room_id, partial) do
+    {:ok, last_partial} = EtsHelper.get_delta_recent_partial(room_id)
+
+    unless is_partial_ready(partial, last_partial) do
+      wait_for_manifest_ready(room_id, partial, :delta_manifest)
+    end
+
+    EtsHelper.get_delta_manifest(room_id)
+  end
+
+  ###
+  ### STORAGE API
+  ###
+
+  @spec update_recent_partial(Room.id(), partial()) :: :ok
+  def update_recent_partial(room_id, partial) do
+    GenServer.cast(registry_id(room_id), {:update_recent_partial, partial, :manifest})
+  end
+
+  @spec update_delta_recent_partial(Room.id(), partial()) :: :ok
+  def update_delta_recent_partial(room_id, partial) do
+    GenServer.cast(registry_id(room_id), {:update_recent_partial, partial, :delta_manifest})
+  end
+
+  ###
+  ### MANAGMENT API
+  ###
+
+  def start(%{room_id: room_id}) do
+    GenServer.start(__MODULE__, %{room_id: room_id, room_pid: self()}, name: registry_id(room_id))
+  end
+
+  def stop(room_id) do
+    GenServer.cast(registry_id(room_id), :shutdown)
+  end
+
+  @impl true
+  def init(%{room_id: room_id, room_pid: room_pid}) do
+    Process.monitor(room_pid)
+    {:ok, %__MODULE__{room_id: room_id, room_pid: room_pid}}
+  end
+
+  @impl true
+  def handle_cast({:update_recent_partial, last_partial, manifest}, state) do
+    status = Map.fetch!(state, manifest)
+    state = Map.put(state, :manifest, update_status(status, last_partial))
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_cast({:is_partial_ready, partial, from, manifest}, state) do
+    state =
+      state
+      |> Map.fetch!(manifest)
+      |> handle_is_partial_ready(partial, from)
+      |> then(&Map.put(state, manifest, &1))
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_cast(:shutdown, state) do
+    {:stop, :normal, state}
+  end
+
+  @impl true
+  def terminate(_reason, %{room_id: room_id}) do
+    EtsHelper.remove_room(room_id)
+  end
+
+  @impl true
+  def handle_info({:DOWN, _ref, :process, pid, _reason}, %{room_pid: pid} = state) do
```

Comment:
Shouldn't `pid` be pinned here? Or used in a `when` clause?

Reply:
it checks whether they are the same or not